extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClientTest.java

```diff
@@ -20,217 +21,392 @@
-                        .isVaultApiHealthStandbyOk(false)
-                        .vaultToken(vaultToken)
-                        .timeout(TIMEOUT)
-                        .build();
-
-
-        var vaultClient = new HashicorpVaultClient(config, edcClientMock, OBJECT_MAPPER);
-        var response = mock(Response.class);
-        var body = mock(ResponseBody.class);
-        var payload = new GetEntryResponsePayload();
-
-        when(edcClientMock.execute(any(Request.class))).thenReturn(response);
-        when(response.code()).thenReturn(200);
-        when(response.body()).thenReturn(body);
-        when(body.string()).thenReturn(payload.toString());
-
-        // invoke
-        var result = vaultClient.getSecretValue(KEY);
-
-        // verify
-        assertNotNull(result);
-        verify(edcClientMock).execute(argThat(request -> request.method().equalsIgnoreCase("GET") &&
-                request.url().encodedPath().contains(CUSTOM_SECRET_PATH + "/data") &&
-                request.url().encodedPathSegments().contains(KEY)));
+    void beforeEach() {
+        reset(monitor);
     }
 
-    @Test
-    void setSecretValue() throws IOException {
-        // prepare
-        var vaultUrl = "https://mock.url";
-        var vaultToken = UUID.randomUUID().toString();
-        var secretValue = UUID.randomUUID().toString();
-        var hashicorpVaultClientConfig =
-                HashicorpVaultClientConfig.Builder.newInstance()
-                        .vaultUrl(vaultUrl)
-                        .vaultApiSecretPath(CUSTOM_SECRET_PATH)
-                        .vaultApiHealthPath(HEALTH_PATH)
-                        .isVaultApiHealthStandbyOk(false)
-                        .vaultToken(vaultToken)
-                        .timeout(TIMEOUT)
-                        .build();
-
-        var vaultClient = new HashicorpVaultClient(hashicorpVaultClientConfig, edcClientMock, OBJECT_MAPPER);
-        var payload = new CreateEntryResponsePayload();
-
-        var call = mock(Call.class);
-        var response = mock(Response.class);
-        var body = mock(ResponseBody.class);
-
-        when(edcClientMock.execute(any(Request.class))).thenReturn(response);
-        when(call.execute()).thenReturn(response);
-        when(response.code()).thenReturn(200);
-        when(response.body()).thenReturn(body);
-        when(body.string()).thenReturn(payload.toString());
-
-        // invoke
-        var result = vaultClient.setSecret(KEY, secretValue);
-
-        // verify
-        assertNotNull(result);
-        verify(edcClientMock).execute(argThat(request -> request.method().equalsIgnoreCase("POST") &&
-                request.url().encodedPath().contains(CUSTOM_SECRET_PATH + "/data") &&
-                request.url().encodedPathSegments().contains(KEY)));
+    @Nested
+    class HealthCheck {
+        @Test
+        void returnsSuccessfulResult_whenHealthCheckCouldBePerformed() throws IOException {
```

Comment:
please read our guidelines on how to name test methods: https://github.com/eclipse-edc/docs/blob/main/developer/contributing/coding-principles.md#v-testing